### PR TITLE
ITL: add all businessprocess module check parameters

### LINKED ISSUE #9107: CheckCommand "icingacli-businessprocess" adding new arguments to icingacli.conf
Some new arguments are available for usage in "icingacli-businenessprocess" and should be added to the default command-definition:

/usr/share/icinga2/include/plugins-contrib.d/icingacli.conf:

```
object CheckCommand "icingacli-businessprocess" {
        import "icingacli"

        command += [ "businessprocess", "process", "check" ]

        arguments = {
                "--config" = {
                        value = "$icingacli_businessprocess_config$"
                        description = "Configuration file containing your business process without file extension"
                }
                "--details" = {
                        set_if = "$icingacli_businessprocess_details$"
                        description = "Get details for root cause analysis"
                }
                "--ack-is-ok" = {
                        set_if = "$icingacli_businessprocess_ackisok$"
                        description = "set state to OK if node is acknowledged"
                }
                "--downtime-is-ok" = {
                        set_if = "$icingacli_businessprocess_downtimeisok$"
                        description = "set state to OK if maintenance is set for node"
                }
                "--blame" = {
                        set_if = "$icingacli_businessprocess_blame$"
                        description = "only showing nodes that have the same state as the process itself"
                }
                "--root-cause" = {
                        set_if = "$icingacli_businessprocess_rootcause$"
                        description = "shows only the path that causes a problem"
                }
                "--state-type" = {
                        value = "$icingacli_businessprocess_statetype$"
                        description = "Define which state type to look at. Could be either soft or hard, overrides an eventually configured default"
                }
                "--process" = {
                        value = "$icingacli_businessprocess_process$"
                        description = "Business process to monitor"
                        skip_key = true
                        required = true
                        order = -1
                }
        }

        vars.icingacli_businessprocess_details = false
        vars.icingacli_businessprocess_ackisok = false
        vars.icingacli_businessprocess_downtimeisok = false
        vars.icingacli_businessprocess_blame = false
        vars.icingacli_businessprocess_rootcause = false
}
```

### PR DESCRIPTION
**Pull request by log1-c**
closes #9110
closes #9107

resolves #9103

### CODE REVIEW

**Comment by bobapple** [comment]
@cla-bot check

File: itl/plugins-contrib.d/icingacli.conf [hunk 1]
```diff
@@ -18,20 +30,33 @@ object CheckCommand "icingacli-businessprocess" {
...
 			order = -1
 		}
+		"--root-cause" = {
+			set_if = "$icingacli_businessprocess_root_cause$"
+			description = "Used in combination with --blame. Only shows the path of the nodes which are responsible for the state of the business process"
+		}
+		"--state-type" = {
+			value = "$icingacli_businessprocess_statetype$"
+			description = "Define which state type to look at. Could be either soft or hard, overrides an eventually configured default"
+		}
```
Comment (Al2Klimov) [inline itl/plugins-contrib.d/icingacli.conf:51]: Why did you move this?

Comment (log1-c) [reply itl/plugins-contrib.d/icingacli.conf:51]: to have it all in alphabetical order 😅 *monk*

**Comment by Al2Klimov** [comment]
Note: must be merged with squash-and-merge.

**Comment by tectumopticum** [comment]
naming of the defined variables should be consistent with the ones already in use, e.g.
"--state-type" = "$icingacli_businessprocess_statetype$"
I suggest e.g.
"--ack-is-ok" should be "$icingacli_businessprocess_ackisok" instead of "$icingacli_businessprocess_ack_is_ok" and so on (s. PR  #9110)

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Good point!

While you're on it, undo the alphabetical sorting of already not alphabetically sorted params.

**Comment by log1-c** [comment]
changes done :)

**Review by yhabteab — CHANGES_REQUESTED** [review]
Can you also reorder argument `state-type` after the `root-cause` arg, shall be sorted alphabetically :) and update the [documentation](https://github.com/Icinga/icinga2/blob/master/doc/10-icinga-template-library.md#business-process-) accordingly.

**Comment by log1-c** [comment]
one doesn't want alphabetical sorting, the other does, not sure what is correct now, but here you go.
Docs updated as well.

**Review by yhabteab — DISMISSED** [review]
Sorry, didn't see it, that @Al2Klimov  change requested you not to do that! But looks fine to me now.

File: doc/10-icinga-template-library.md [hunk 2]
```diff
@@ -2839,6 +2839,11 @@ icingacli_businessprocess_process         | **Required.** Business process to mo
 icingacli_businessprocess_config          | **Optional.** Configuration file containing your business process without file extension.
 icingacli_businessprocess_details         | **Optional.** Get details for root cause analysis. Defaults to false.
 icingacli_businessprocess_statetype       | **Optional.** Define which state type to look at, `soft` or `hard`. Overrides the default value inside the businessprocess module, if configured.
+icingacli_businessprocess_ackisok         | **Optional.** Treat acknowledged hosts/services always as UP/OK.
+icingacli_businessprocess_blame           | **Optional.** Show problem details as a tree reduced to the nodes which have the same state as the business process
+icingacli_businessprocess_colors          | **Optional.** Show colored output
+icingacli_businessprocess_downtimeisok    | **Optional.** Treat hosts/services in downtime always as UP/OK.
+icingacli_businessprocess_rootcause       | **Optional.** Used in combination with *icingacli_businessprocess_blame*. Only shows the path of the nodes which are responsible for the state of the business process
```
Comment (Al2Klimov) [inline doc/10-icinga-template-library.md:2846]: Some sentences miss full stops.

Comment (Al2Klimov) [inline doc/10-icinga-template-library.md:2846]: _path_ is singular, but _are_ implies plural.

File: itl/plugins-contrib.d/icingacli.conf [hunk 3]
```diff
@@ -18,20 +30,34 @@ object CheckCommand "icingacli-businessprocess" {
...
 			order = -1
 		}
+		"--root-cause" = {
+			set_if = "$icingacli_businessprocess_rootcause$"
+			description = "Used in combination with --blame. Only shows the path of the nodes which are responsible for the state of the business process"
+		}
+		"--state-type" = {
+			value = "$icingacli_businessprocess_statetype$"
+			description = "Define which state type to look at. Could be either soft or hard, overrides an eventually configured default"
+		}
```
Comment (Al2Klimov) [inline itl/plugins-contrib.d/icingacli.conf:51]: Same stuff here. (Actually the reverse regarding full stops.)

File: itl/plugins-contrib.d/icingacli.conf [hunk 4]
```diff
@@ -18,20 +30,34 @@ object CheckCommand "icingacli-businessprocess" {
...
 			description = "Get details for root cause analysis"
 		}
-		"--state-type" = {
-			value = "$icingacli_businessprocess_statetype$"
-			description = "Define which state type to look at. Could be either soft or hard, overrides an eventually configured default"
-		}
+		"--downtime-is-ok" = {
+			set_if = "$icingacli_businessprocess_downtimeisok$"
+			description = "Treat hosts/services in downtime always as UP/OK."
+		}
```
Comment (Al2Klimov) [inline itl/plugins-contrib.d/icingacli.conf:36]: Redundant trailing whitespace.

File: itl/plugins-contrib.d/icingacli.conf [hunk 5]
```diff
@@ -18,20 +30,34 @@ object CheckCommand "icingacli-businessprocess" {
...
 	}
 
+	vars.icingacli_businessprocess_ackisok = false	
+	vars.icingacli_businessprocess_blame = false
+	vars.icingacli_businessprocess_colors = false
 	vars.icingacli_businessprocess_details = false
+	vars.icingacli_businessprocess_downtimeisok = false
+	vars.icingacli_businessprocess_rootcause = false
+	vars.icingacli_businessprocess_statetype = false
```
Comment (Al2Klimov) [inline itl/plugins-contrib.d/icingacli.conf:60]: At least the last line shall not be there.

File: itl/plugins-contrib.d/icingacli.conf (hunk 3 above)
Comment (Al2Klimov) [inline itl/plugins-contrib.d/icingacli.conf:51]: Oh, and while you’re on it: please put this one back. (Yonas has already apologised for giving you an instruction which conflicted with mine.)

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: itl/plugins-contrib.d/icingacli.conf (hunk 3 above)
Comment (log1-c) [reply itl/plugins-contrib.d/icingacli.conf]: moved back again

File: itl/plugins-contrib.d/icingacli.conf (hunk 5 above)
Comment (log1-c) [reply itl/plugins-contrib.d/icingacli.conf]: correct, got confused by the whole parameter moving

File: itl/plugins-contrib.d/icingacli.conf (hunk 4 above)
Comment (log1-c) [reply itl/plugins-contrib.d/icingacli.conf]: whitespace removed

File: doc/10-icinga-template-library.md (hunk 2 above)
Comment (log1-c) [reply doc/10-icinga-template-library.md]: corrected
(was c&p from the BP modules command output)

File: itl/plugins-contrib.d/icingacli.conf (hunk 3 above)
Comment (log1-c) [reply itl/plugins-contrib.d/icingacli.conf]: removed full stops like in the other parameters

File: doc/10-icinga-template-library.md (hunk 2 above)
Comment (log1-c) [reply doc/10-icinga-template-library.md]: corrected

Comment (Al2Klimov) [reply doc/10-icinga-template-library.md]: Actually this one...

File: itl/plugins-contrib.d/icingacli.conf (hunk 3 above)
Comment (Al2Klimov) [reply itl/plugins-contrib.d/icingacli.conf]: ... applies here as well.

Comment (log1-c) [reply itl/plugins-contrib.d/icingacli.conf:51]: fixed. got overlooked yesterday. one should do changes with both eyes on the screen 🤯

**Review by Al2Klimov — APPROVED** [review]

**Review by yhabteab — APPROVED** [review]
